### PR TITLE
chore: run tests in the buildkit itself

### PR DESCRIPTION
**Pull request by smira**
This relies on two PRs to the buildkit, which aren't merged yet, so I
had to do some overrides to apply them:

* https://github.com/moby/buildkit/pull/1081
* https://github.com/moby/buildkit/pull/1085

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Comment by smira** [comment]
Forgot to remove `dev-test` target, I'll do it in the next push, want to see if it works in drone

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 TOOLS ?= autonomy/tools:b4e3778
 
 # TODO(andrewrynhard): Move this logic to a shell script.
-BUILDKIT_VERSION ?= v0.5.0
+BUILDKIT_VERSION ?= master@sha256:455f06ede03149051ce2734d9639c28aed1b6e8b8a0c607cb813e29b469a07d6
```
Comment (andrewrynhard) [inline Makefile]: I can update our current deployment to match this.

Comment (andrewrynhard) [reply Makefile]: Updated and it looks like this works!

Comment (smira) [reply Makefile]: oh, that's awesome!

let me rebase and squash it. still waiting on dockerfile PR, and we can update `Dockerfile` slug back to upstream

**Pull request merged**